yahp/utils/type_helpers.py

```diff
@@ -412,3 +413,72 @@ def is_none_like(x: Any, *, allow_list: bool) -> bool:
     if allow_list and isinstance(x, (tuple, list)) and len(x) == 1:
         return is_none_like(x[0], allow_list=allow_list)
     return False
+
+
+def get_type_json_schema(f_type: Any):
```

Comment:
Rewriting this function using HparamsType will help cover more of the edge cases with typing annotations.

```suggestion
def get_type_json_schema(f_type: HparamsType):
```